Added and tweaked testing scenarios (#330)
microsoft/devhomegithubextension: TestingScenarios.md

## Testing Scenarios

These are the testing scenarios that need to be validated before shipping a new release.  When an automated test is added, please remove it from the below lists.

### Widgets

1. For each widget: user can configure the widget immediately after first pin

> 2. For each widget: user can customize (reconfigure) the widget after its been previously configured and pinned

Reviewer: ```suggestion
2. For each widget: user can customize (reconfigure) the widget after it was initially configured
```

> 2. For each widget: user can customize (reconfigure) the widget after its been previously configured and pinned

Reviewer: Nice thing about markdown is you can just use 1. for everything and it will format it correctly in not-raw mode. Useful if you add items in the middle, then you don't have to go renumber everything. I'd leave the 1s. Not sure if the script expects them as well.

Reviewer: Updated
